### PR TITLE
RFC 59 - show studies for which a user is not authorized

### PR DESCRIPTION
**Pull request by dianab0**
Rfc 59 (https://docs.google.com/document/d/1NkNhBG1sk0ZvCOwk3XtiuRT9a_tKkFCgXC9I3XT-Ye4/edit#heading=h.6c2l76kgw1ta)
Physical studies that are not authorized are grayed out, have a gray info icon with no tooltip and a lock icon at then end. This behavior can be enabled/disabled from portal.properties (by default false). Also the global message that appears when hover over lock icon can be configured in portal.properties as well.

Screenshot from 2021-08-17 12-24-03

Backend PR is https://github.com/cBioPortal/cbioportal/pull/8847

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -4120,7 +4120,14 @@ export class StudyViewPageStore
             return _.reduce(
                 this.studyIds,
                 (acc: CancerStudy[], next) => {
-                    if (everyStudyIdToStudy[next]) {
+                    if (
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:4123]: lets put this condition in a helper function

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -4120,7 +4120,14 @@ export class StudyViewPageStore
...
+                        everyStudyIdToStudy[next] &&
+                        (!AppConfig.serverConfig
+                            .skin_show_unauthorized_studies ||
+                            (AppConfig.serverConfig
+                                .skin_show_unauthorized_studies &&
+                                everyStudyIdToStudy[next].isAuthorized == true))
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts:4129]: use ===

Comment (alisman) [reply src/pages/studyView/StudyViewPageStore.ts:4129]: == in javascript means type coercion.  it's just a good practice to use === as a rule

File: src/shared/components/query/studyList/StudyList.tsx [hunk 3]
```diff
@@ -367,27 +372,37 @@ export default class StudyList extends QueryStoreComponent<
...
-                                    isVirtualStudy={this.store.isVirtualStudy(
-                                        study.studyId
-                                    )}
-                                    mouseEnterDelay={0}
-                                    placement="top"
-                                >
-                                    <a>{content}</a>
-                                </StudyTagsTooltip>
-                            );
+                            if (
```
Comment (alisman) [inline src/shared/components/query/studyList/StudyList.tsx:375]: again, lets wrap in a helper function.

**Comment by dianab0** [comment]
Thank you for review @alisman. I updated the code. I added the helper function in both StudyViewPageStore.ts and StudyList.tsx. However in the StudyList we may remove it the future if we'll implement, as per RFC, the retrieval of info icon data for unauthorized studies too.

**Comment by dianab0** [comment]
@alisman sorry, I forgot to push the commit containing the updates after review. It is pushed now.

File: src/shared/components/query/studyList/StudyList.tsx [hunk 4]
```diff
@@ -66,6 +66,14 @@ export default class StudyList extends QueryStoreComponent<
         if (this.props.showSelectedStudiesOnly) this._view = this.view;
     }
 
+    isQueriedStudyAuthorized = (study: CancerStudy): boolean => {
```
Comment (alisman) [inline src/shared/components/query/studyList/StudyList.tsx:69]: can we put this in utils so we don't define it twice?

**Pull request merged**